02-microarray/pathway-analysis_microarray_02_ora.Rmd

```diff
@@ -0,0 +1,476 @@
+# We will need this so we can use the pipe: %>%
+library(magrittr)
+```
+
+## Import data 
+
+We will read in the differential expression results we will download from online.
+These results are from a zebrafish microarray experiment we used for [differential expression analysis for two groups](https://alexslemonade.github.io/refinebio-examples/02-microarray/differential-expression_microarray_02_2-groups.html) using [`limma`](https://bioconductor.org/packages/release/bioc/html/limma.html) [@Ritchie2015].
+The table contains Ensembl gene IDs, log fold-changes for each group, and adjusted p-values (FDR in this case).
+We can identify upregulated genes by filtering these results and use this list as input to ORA.
+
+Instead of using the URL below, you can use a file path to a TSV file with your desired gene list results.
+
+```{r}
+# Define the url to your differential expression results file
+dge_url <- "https://refinebio-examples.s3.us-east-2.amazonaws.com/02-microarray/results/GSE71270/GSE71270_limma_results.tsv"
+```
+
+Read in the file that has differential expression results. 
+Here we are using the URL we set up above, but this can be a local file path instead _i.e._ you can replace `dge_url` in the code below with a path to file you have on your computer like: `file.path("results", "GSE71270_limma_results.tsv")`.
+
+```{r}
+# Read in the contents of your differential expression results file
+dge_df <- readr::read_tsv(dge_url) # `dge_url` can be replaced with a file path to a TSV file with your desired gene list results
+```
+
+`read_tsv()` can read TSV files online and doesn't necessarily require you download the file first. 
+Let's take a look at what these contrast results from the differential expression analysis look like. 
+
+```{r}
+dge_df
+```
+
+## Getting familiar with `clusterProfiler`'s options
+
+Let's take a look at what organisms the package supports.
+
+```{r}
+msigdbr_species()
+```
+
+The data we're interested in here comes from zebrafish samples, so we can obtain just the gene sets relevant to _D. rerio_ with the `species` argument to `msigdbr()`.
+
+```{r}
+hs_msigdb_df <- msigdbr(species = "Danio rerio")
+```
+
+MSigDB contains [8 different gene set collections](https://www.gsea-msigdb.org/gsea/msigdb/collections.jsp) [@Subramanian2005].
+
+    H: hallmark gene sets
+    C1: positional gene sets
+    C2: curated gene sets
+    C3: motif gene sets
+    C4: computational gene sets
+    C5: GO gene sets
+    C6: oncogenic signatures
+    C7: immunologic signatures
+
+In this example, we will use pathways that are gene sets considered to be "canonical representations of a biological process compiled by domain experts" and are a subset of `C2: curated gene sets` [@Subramanian2005].
+
+Specifically, we will use the [KEGG (Kyoto Encyclopedia of Genes and Genomes)](https://www.genome.jp/kegg/) pathways [@Kanehisa2000]. 
+
+First, let's take a look at what information is included in this data frame. 
+
+```{r}
+head(hs_msigdb_df)
```

Comment:
Also related to [a comment on #246](https://github.com/AlexsLemonade/refinebio-examples/pull/246#discussion_r502661357) the tables in the html document are not rendering in the most informative matter. In that PR I added the `df_print:paged` option [at the top of the notebook](https://github.com/AlexsLemonade/refinebio-examples/pull/246/files#diff-6e1a150e702a9b814aaf52aa272f45dadb468d003ccbfcdbc6be1da2ccd5cb55R10) and in the [`render-notebooks.R` script](https://github.com/AlexsLemonade/refinebio-examples/pull/246/files#diff-b7c02b6fd7d59024caba557c65d1602a77db88657e463653d40ec16227740484R115).

We may want to incorporate this method here (and throughout the notebook where a table is being displayed) or a better alternative if you have any ideas.

Reply:
I like this idea. And since it appears you've added it to the render script, there' not a need for me to add it here (once #246 is merged, I'll merge it here and we'll have it). 

However, I did comment on there a discussion we should have about if we want to include it in the individual headers (if we do we should probably put it in all headers since we do a lot of printing out data frames).